### PR TITLE
If it is https make sure we send secure

### PR DESCRIPTION
**Pull request by harikt**
Hi @ryancramerdesign , 

I have added a small piece to make sure when the value is https it is set secure, and if not no . `setcookie(session_name() . '_challenge', $challenge, time()+60*60*24*30, '/', null, (bool) $this->config->https, true);`

As our coding standards are a bit different the whitespace seems eaten up by my editor. Feel free to take the code and commit so we can make sure the cookies are secure.

Thank you.

### CODE REVIEW

**Comment by harikt** [comment]
Further you can check with w=1 , https://github.com/ryancramerdesign/ProcessWire/pull/1264/files?w=1

Thanks

**Comment by harikt** [comment]
By the way where comes the wire session cookie ? I was looking where it is set from.

**Comment by LostKobrakai** [comment]
> By the way where comes the wire session cookie ?

[Here](https://github.com/ryancramerdesign/ProcessWire/blob/e577e10b3e6cfdc97f1909d834ddc7f002853eae/index.php#L177) it's set and [here](https://github.com/ryancramerdesign/ProcessWire/blob/master/wire/core/Session.php#L125) it's created.

**Comment by harikt** [comment]
@LostKobrakai awesome catch. Thank you. Will make a further update.

**Comment by harikt** [comment]
Seems like even if I made it set secure seems something is wrong for it seems not setting :( .

File: index.php [hunk 1]
```diff
@@ -177,7 +177,10 @@ function ProcessWireBootConfig() {
 	ini_set('session.use_cookies', true); 
 	ini_set('session.use_only_cookies', 1);
 	ini_set('session.cookie_httponly', 1); 
-	ini_set('session.gc_maxlifetime', $config->sessionExpireSeconds); 
+	ini_set('session.gc_maxlifetime', $config->sessionExpireSeconds);
+	if ($config->https) {
```
Comment (harikt) [inline index.php]: at-least this information is not getting here :-/ .

**Comment by ryancramerdesign** [comment]
Thanks @harikt this seems like a good addition. I have added it and it will appear in this week's commits. Regarding the index.php change, the value of $config->https has not yet been set (PW sets it in /wire/core/ProcessWire.php), so I've added the following line above your if($config->https) to resolve it:
```````
$config->https = (!empty($_SERVER['HTTPS']) && strtolower($_SERVER['HTTPS']) == 'on') 
  || (isset($_SERVER['SERVER_PORT']) && $_SERVER['SERVER_PORT'] == 443);
```````

**Comment by harikt** [comment]
Hey @ryancramerdesign , 

Thank you, is the commits done to dev branch ? May be if you can link to the specific commit it will help me to check. I had a look, didn't find it though.

Thank you

**Comment by LostKobrakai** [comment]
It's possible that it's not on GitHub by now. I'd take a look on Friday, when everything for the week get's finished up.

**Comment by isellsoap** [comment]
@harikt Can this PR be closed? If yes, please feel free to do so. :-)

**Comment by harikt** [comment]
@isellsoap PR's are about to be merged not closed. But the process is a bit different here. 

Closing for it seems already done :) .

Thank you.

**Pull request closed without merging**